### PR TITLE
cocoa: add an option to disable the native macOS fullscreen

### LINKED ISSUE #4014: Disable macOS native full screen?
Hi. I just updated to version v0.23.0. I have two displays and whenever I enter full screen since the update, mpv seems to use the native macOS full screen feature and dims out my second display. Is there a way to revert back to the old behaviour?

I'm using the latest El Capitan, fwiw. Any help would be much appreciated.

### PR DESCRIPTION
**Pull request by Akemi**
rather self explanatory.

i tried to keep the code of the none-native fullscreen as separate as possible, so it is a bit easier to maintain. though it leads to some minor code duplications.

### CODE REVIEW

**Comment by tonycpsu** [comment]
Works great for me (MacOS 10.2.5).  Thanks!

**Comment by tonycpsu** [comment]
One minor bug notice that the position of the window isn't restored when going from windowed to fullscreen and then back.  The X position seems fine, but the Y position is pretty far off.

**Comment by Akemi** [comment]
that should fix it.

**Comment by patrickchoi** [comment]
Is this possible to put in the config file? I tried putting no-native-fs in there but it didn't do anything

**Comment by tonycpsu** [comment]
@patrickchoi : try `native-fs=no`

**Comment by Akemi** [comment]
also make sure to use a more recent version of mpv build from master. it's not in the 0.26 release and will be in 0.27.

**Comment by albertnetymk** [comment]
Works great. Thanks for the work.

**Comment by porg** [comment]
Thanks for this feature-toggle!

I prefer the user experience of instant switching between windowed and fullscreen mode, which is now again possible via `native-fs=no` in my config file.

One question out of curiosity and regrding performance: Is `native-fs=no` only choosing between different transition effects between windowed and fullscreen mode, or does it choose between entirely different mechanisms on how to draw fullscreen?

In other words: Does `native-fs=no` carry any performance penalties or gains compared to macOS's owns fullscreen drawing mechanism? The instant switching surely is more primitive than the animated slide in, but that does not necessarily translate into more performant, as native code/APIs are often more efficient than 3rd party implementations. Thatswhy my curiosity. Thanks.

**Comment by Hrxn** [comment]
If I recall it correctly, it should not make any difference with regard to performance.

**Comment by Akemi** [comment]
usually it shouldn't have any performance difference, and if then it's most likely another problem related to #2392. that problem should be 'fixed' too, as far as possible that is.

the drawing code is the same. basically and simplified the differences between both fullscreen modes are just some different window flags.

**Comment by porg** [comment]
Thanks for the information!

Knowing that this comes at no performance losses lets me use that feature with even more pleasure.

Thanks for your good work!
